runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java

```diff
@@ -102,32 +102,41 @@ private void translateKafkaInput(
       FlinkStreamingPortablePipelineTranslator.StreamingTranslationContext context) {
     RunnerApi.PTransform pTransform = pipeline.getComponents().getTransformsOrThrow(id);
 
-    final String topic;
-    final Properties properties = new Properties();
-    ObjectMapper mapper = new ObjectMapper();
+    final Map<String, Object> params;
     try {
-      Map<String, Object> params =
-          mapper.readValue(pTransform.getSpec().getPayload().toByteArray(), Map.class);
-
-      Preconditions.checkNotNull(topic = (String) params.get("topic"), "'topic' needs to be set");
-
-      Map<?, ?> consumerProps = (Map) params.get("properties");
-      Preconditions.checkNotNull(consumerProps, "'properties' need to be set");
-      properties.putAll(consumerProps);
+      ObjectMapper mapper = new ObjectMapper();
+      params = mapper.readValue(pTransform.getSpec().getPayload().toByteArray(), Map.class);
     } catch (IOException e) {
       throw new RuntimeException("Could not parse KafkaConsumer properties.", e);
     }
 
+    final String topic;
+    Preconditions.checkNotNull(topic = (String) params.get("topic"), "'topic' needs to be set");
+
+    Map<?, ?> consumerProps = (Map) params.get("properties");
+    final Properties properties = new Properties();
+    Preconditions.checkNotNull(consumerProps, "'properties' need to be set");
+    properties.putAll(consumerProps);
+
     logger.info("Kafka consumer for topic {} with properties {}", topic, properties);
 
-    DataStreamSource<WindowedValue<byte[]>> source =
+    FlinkKafkaConsumer011<WindowedValue<byte[]>> kafkaSource =
+        new FlinkKafkaConsumer011<>(topic, new ByteArrayWindowedValueSchema(), properties);
+
+    kafkaSource.setStartFromLatest();
```

Comment:
Do you want to integrate the offset fix here if it is not too much work?

Reply:
That's a separate change.